plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/FlushMetadataProcedure.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.procedure;
+
+import io.trino.plugin.deltalake.transactionlog.TransactionLogAccess;
+import io.trino.plugin.hive.metastore.cache.CachingHiveMetastore;
+import io.trino.plugin.hive.metastore.procedure.FlushHiveMetastoreCacheProcedure;
+import io.trino.spi.TrinoException;
+import io.trino.spi.classloader.ThreadContextClassLoader;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.trino.plugin.deltalake.metastore.HiveMetastoreBackedDeltaLakeMetastore.getTableLocation;
+import static io.trino.plugin.deltalake.metastore.HiveMetastoreBackedDeltaLakeMetastore.verifyDeltaLakeTable;
+import static io.trino.plugin.hive.HiveErrorCode.HIVE_METASTORE_ERROR;
+import static io.trino.spi.StandardErrorCode.INVALID_PROCEDURE_ARGUMENT;
+import static java.util.Objects.requireNonNull;
+
+public class FlushMetadataProcedure
+        extends FlushHiveMetastoreCacheProcedure
+{
+    private final TransactionLogAccess transactionLogAccess;
+
+    @Inject
+    public FlushMetadataProcedure(Optional<CachingHiveMetastore> cachingHiveMetastore, TransactionLogAccess transactionLogAccess)
+    {
+        super(cachingHiveMetastore);
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+    }
+
+    @Override
+    public void flushMetadataCache(String schemaName, String tableName, List<String> partitionColumns, List<String> partitionValues, List<String> partitionColumn, List<String> partitionValue)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doFlushMetadataCache(Optional.ofNullable(schemaName), Optional.ofNullable(tableName));
+        }
+        super.flushMetadataCache(schemaName, tableName, partitionColumns, partitionValues, partitionColumn, partitionValue);
+    }
+
+    private void doFlushMetadataCache(Optional<String> schemaName, Optional<String> tableName)
+    {
+        // TODO: Allow flushing TransactionLogAccess's cache even when CachingHiveMetastore is disabled
+        CachingHiveMetastore cachingHiveMetastore = this.cachingHiveMetastore
+                .orElseThrow(() -> new TrinoException(HIVE_METASTORE_ERROR, "Cannot flush, metastore cache is not enabled"));
+        if (schemaName.isEmpty() && tableName.isEmpty()) {
+            transactionLogAccess.flushCache();
```

Comment:
Can we have an integration test covering this path as well?
At the moment there is only a test for invalidating the cache of a specific table in `TestDeltaLakeActiveFilesCache`.